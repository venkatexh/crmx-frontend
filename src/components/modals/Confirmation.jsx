import "../../sass/components/modals/confirmation.component.scss";
import {toggleConfirmationModal} from "../../redux/actions/utility/toggleConfirmationModal";
import {useDispatch} from "react-redux";

const Confirmation = ({payload}) => {
  const {message, trueText, falseText, action, type} = payload;
  const dispatch = useDispatch();

  const closeModal = () => {
    dispatch(toggleConfirmationModal(null))
  }
  return (
    <div className={'confirmationModalContainer'}>
      <div className={'confirmationModal'}>
        <div className={'confMessage'}>{message}</div>
        <div className={'buttonsContainer'}>
          <button onClick={action}
                  className={`${type === 'error' ? 'errorButton' : 'successButton'} confButton`}>{trueText}</button>
          <button onClick={closeModal} className={'noButton confButton'}>{falseText}</button>
        </div>
      </div>
    </div>
  )
}

export default Confirmation;