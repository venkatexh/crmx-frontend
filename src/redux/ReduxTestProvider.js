import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
