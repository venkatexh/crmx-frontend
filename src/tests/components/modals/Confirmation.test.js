import { shallow } from "enzyme";
import React from "react";
import Confirmation from "../../../components/modals/Confirmation";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("should be confirmation modal component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <Confirmation
          payload={{
            message: "sure wantto do this?",
            trueText: "yes",
            falseText: "no",
            action: () => {},
            type: "success type",
          }}
        />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
