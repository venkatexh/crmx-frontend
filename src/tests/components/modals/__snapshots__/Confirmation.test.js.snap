// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be confirmation modal component snapshot 1`] = `ShallowWrapper {}`;
